src/vs/workbench/browser/window.ts

```diff
@@ -231,13 +233,13 @@ export class BrowserWindow extends Disposable {
 							);
 						}
 
-						await this.dialogService.prompt({
+						await this.hostService.withSilentShutdown?.(() => this.dialogService.prompt({
```

Comment:
I find it a bit strange to call a method with `?` where we do not know if it is defined because it will not be called if not defined and thus the prompt would never show?